### PR TITLE
Support specifying an object as unmanaged.

### LINKED ISSUE #376: Support to annotate objects as unmanaged.
There might be cases that we have to pause operator reconciliations

- to do manual operation and maintenance
- to debug the behaviour of the cluster and the operator
- to avoid operator bugs

We could use a certain annotation as a switch to pause reconciliations and check the switch in the control loop.

### PR DESCRIPTION
**Pull request by MarvinCai**
resolves #376

### Motivation
Adding annotation compute.functionmesh.io/managed: false should make an object unmanged.

### Modifications
Controller will check for the annotation in reconcile loop and skip unmanaged objects.

### Verifying this change

- [x] Make sure that the change passes the CI checks.

This change is a trivial rework / code cleanup without any test coverage.

### Documentation

- [x] `doc-required` 
  Add doc about how to make an object unmanaged.

### CODE REVIEW

**Review by freeznet — APPROVED** [review]
lgtm, @MarvinCai can you fix the CI with `goimports` errors? thanks.

**Comment by freeznet** [comment]
@nlu90 @tpiperatgod @jiangpengcheng PTAL when you have time, thanks.

**Review by tuteng — APPROVED** [review]

**Review by fantapsody — APPROVED** [review]

**Review by jiangpengcheng — APPROVED** [review]

File: controllers/spec/common.go [hunk 1]
```diff
@@ -51,9 +51,9 @@ const (
 	PackageNameSinkPrefix     = "sink://"
 	PackageNameSourcePrefix   = "source://"
 
-	AnnotationPrometheusScrape  = "prometheus.io/scrape"
-	AnnotationPrometheusPort    = "prometheus.io/port"
-	AnnotationAppliedConfigHash = "config.functionmesh.io/applied"
```
Comment (jiangpengcheng) [inline controllers/spec/common.go:56]: this annotation won't be used?

Comment (freeznet) [reply controllers/spec/common.go:56]: correct

**Pull request merged**